simpleline/render/renderer.py

```diff
@@ -0,0 +1,382 @@
+# Class handling rendering of the screens to console.
+#
+# Copyright (C) 2017  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Author(s): Jiri Konecny <jkonecny@redhat.com>
+#
+
+import getpass
+import queue
+import sys
+import threading
+
+from simpleline.event_loop import ExitMainLoop
+from simpleline.event_loop.signals import ExceptionSignal, InputReadySignal, RenderScreenSignal, InputScreenSignal
+from simpleline.render import RendererUnexpectedError, INPUT_PROCESSED, INPUT_DISCARDED
+from simpleline.render.prompt import Prompt
+from simpleline.render.screen_stack import ScreenStack, ScreenData, ScreenStackEmptyException
+from simpleline.render.widgets import TextWidget
+
+RAW_INPUT_LOCK = threading.Lock()
+
+
+class Renderer(object):
+
+    def __init__(self, event_loop, renderer_stack=None):
+        """Constructor where you can pass your own renderer stack.
+
+        The ScreenStack will be used automatically if renderer stack will be None.
+
+        :param event_loop: Event loop used for the renderer.
+        :type event_loop: Class based on `simpleline.event_loop.AbstractEventLoop`.
+        :param renderer_stack: Input your own renderer stack if you need to.
```

Comment:
"Input your own renderer stack if you need to." sounds weird to me
Maybe:
"Use custom renderer stack.